src/test/java/mocacong/server/concurrency/CafeConcurrentServiceTest.java

```diff
@@ -0,0 +1,133 @@
+package mocacong.server.concurrency;
+
+import mocacong.server.domain.Cafe;
+import mocacong.server.domain.Member;
+import mocacong.server.domain.Review;
+import mocacong.server.dto.request.CafeRegisterRequest;
+import mocacong.server.dto.request.CafeReviewRequest;
+import mocacong.server.exception.badrequest.AlreadyExistsCafeReview;
+import mocacong.server.exception.badrequest.DuplicateCafeException;
+import mocacong.server.repository.CafeRepository;
+import mocacong.server.repository.MemberRepository;
+import mocacong.server.repository.ReviewRepository;
+import mocacong.server.service.CafeService;
+import mocacong.server.service.MemberService;
+import mocacong.server.service.ServiceTest;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+@ServiceTest
+public class CafeConcurrentServiceTest {
+    @Autowired
+    private CafeService cafeService;
+    @Autowired
+    private MemberService memberService;
+    @Autowired
+    private CafeRepository cafeRepository;
+    @Autowired
+    private MemberRepository memberRepository;
+    @Autowired
+    private ReviewRepository reviewRepository;
+
+    @Test
+    @DisplayName("등록되지 않은 카페를 동시에 여러 번 등록하려 해도 한 번만 등록된다")
+    void saveCafeWithConcurrent() throws InterruptedException {
+        CafeRegisterRequest request = new CafeRegisterRequest("20", "메리네 카페");
+        ExecutorService executorService = Executors.newFixedThreadPool(3);
+        CountDownLatch latch = new CountDownLatch(3);
+        List<Throwable> exceptions = Collections.synchronizedList(new ArrayList<>());
+
+        for (int i = 0; i < 3; i++) {
+            executorService.execute(() -> {
+                try {
+                    cafeService.save(request);
+                } catch (DuplicateCafeException e) {
+                    exceptions.add(e); // 중복 예외를 리스트에 추가
+                }
+                latch.countDown();
+            });
+        }
+        latch.await();
+
+        List<Cafe> actual = cafeRepository.findAll();
+        assertAll(
+                () -> assertThat(exceptions.isEmpty()).isFalse(),
+                () -> assertThat(actual).hasSize(1)
+        );
```

Comment:
아래 방법도 괜찮을 거 같기도 하네요
```suggestion
        assertAll(
                () -> assertThat(exceptions).hasSize(2),
                () -> assertThat(actual).hasSize(1)
        );
```

Reply:
이 방법도 예외가 2번 일어났다는 걸 잘보여주는 거 같아서 좋네요 👍